### DISCUSSION TITLE
Plan parsed from a string

### DISCUSSION THREAD

**Discussion by jdavidrcamacho** [aiplan4eu/unified-planning#522] [Q&A] [answered]
Hi!

While exploring how to parse a  temporal plan from a string using the function parse_plan_string from PDDLReader I noticed that on lines 1744-1748: 

            s_ai = re.match(r"^\s*\(\s*([\w?-]+)((\s+[\w?-]+)*)\s*\)\s*$", line)
            t_ai = re.match(
                r"^\s*(\d+)\s*:\s*\(\s*([\w?-]+)((\s+[\w?-]+)*)\s*\)\s*(\[\s*(\d+)\s*\])?\s*$",
                line,
            )
            
 It is where the function tries to identify if its a sequential plan or a  time triggered plan. 
 
Now my question is if the time triggered string is, for example, "0.0: (action-name param1 param2 ... paramN) [1.5]" , meaning star time and duration are meant to be floats the parse_plan_string does not work because it looks only for integer values. 
Was this implemented on purpose or is it a bug?

**Comment by arbimo** [comment, marked as answer]
This is a bug that was fixed in #516 
The fix will be included in the next release.

**Comment by jdavidrcamacho** [comment]
Thank you!
